gazelle/python/generate.go

```diff
@@ -463,6 +470,20 @@ func hasEntrypointFile(dir string) bool {
 	return false
 }
 
+// hasNameEqualsMain determines if the file contains 'if __name__ == "__main__"'.
+func hasNameEqualsMain(path string) bool {
```

Comment:
I personally think that this new feature needs a feature toggle. There should be a gazelle directive that enables the behaviour.

What is more, sometimes developers add `if __name__` in order to test the script locally but do not intend to create a binary target for others to consume. I wonder if in those cases we should be able to say in the python file:
```
if __name__ == "__main__":  # gazelle: ignore
    main()
```

Reply:
A feature toggle is a great call. Let me look into making that work. As for the ignore, is there any reason you wouldn't want that to get moved to a `py_binary`? Then you could `bazel run` it and test it locally. (If the script is already in a test target, then this change won't move it - it only moves scripts from `py_library` -> `py_binary`, not from `py_test`, because you can already `bazel run` a test.) I guess I'm just curious when you might want an ignore / might want to keep a script out of a `py_binary` target.

Reply:
Let's keep the ignore part out of scope for now. It probably does not need to be supported in the initial version.